pkg/controller/nmstate/nmstate_controller_test.go

```diff
@@ -69,4 +69,32 @@ var _ = Describe("NMstate controller reconcile", func() {
 			Expect(result).To(Equal(reconcile.Result{}))
 		})
 	})
+	Context("node selector is set ", func() {
+		var (
+			request reconcile.Request
+		)
+		BeforeEach(func() {
+			s := scheme.Scheme
+			s.AddKnownTypes(nmstatev1alpha1.SchemeGroupVersion,
+				&nmstatev1alpha1.NMstate{},
+			)
+
+			m := map[string]string{"selector_1": "value_1", "selector_2": "value_2"}
+			nmstate.Spec.NodeSelector = m
+			objs := []runtime.Object{&nmstate}
+
+			// Create a fake client to mock API calls.
+			cl = fake.NewFakeClientWithScheme(s, objs...)
+
+			reconciler.client = cl
+
+			request.Name = existingNMstateName
+		})
+		It("should return empty Result and no error", func() {
+			result, err := reconciler.Reconcile(request)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(result).To(Equal(reconcile.Result{}))
+		})
```

Comment:
We have to retrieve DaemonSets from handler and check node selector is set there.